homeassistant/components/sensor/homematicip.py

```diff
@@ -0,0 +1,418 @@
+        if hasattr(self._device, 'rssiDeviceValue'):
+            attr.update({ATTR_RSSI_DEVICE: self._device.rssiDeviceValue})
+        if hasattr(self._device, 'rssiPeerValue'):
+            attr.update({ATTR_RSSI_PEER: self._device.rssiPeerValue})
+        return attr
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the generic device."""
+        return self._generic_state_attributes()
+
+
+class HomematicipAccesspoint(HomematicipGenericDevice):
+    """Representation of an HomeMaticIP access point."""
+
+    def __init__(self, hass, home):
+        """Initialize the access point sensor."""
+        super().__init__(hass, home, home)
+
+        @callback
+        def event_home(event):
+            """Handle device state changes."""
+            _LOGGER.debug('Event Access Point: %s', self._home.label)
+            self.async_schedule_update_ha_state(True)
+
+        hass.bus.listen(EVENT_HOME_CHANGED, event_home)
+
+    @property
+    def name(self):
+        """Return the name of the access point device."""
+        if self._home.label == '':
+            return 'Access Point Status'
+        return '{} Access Point Status'.format(self._home.label)
+
+    @property
+    def icon(self):
+        """Return the icon of the access point device."""
+        return 'mdi:access-point-network'
+
+    @property
+    def state(self):
+        """Return the state of the access point."""
+        return self._home.dutyCycle
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the access point."""
+        return {
+            ATTR_LABEL: self._home.label,
+            ATTR_ID: self._home.id,
+            ATTR_CONNECTED: self._home.connected,
+            ATTR_DUTYCYCLE: self._home.dutyCycle,
+            ATTR_CURRENT_APVERSION: self._home.currentAPVersion,
+            ATTR_AVAILABLE_APVERSION: self._home.availableAPVersion,
+            ATTR_TIMEZONE_ID: self._home.timeZoneId,
+            ATTR_PIN_ASSIGNED: self._home.pinAssigned,
+            ATTR_UPDATE_STATE: self._home.updateState,
+            ATTR_APEXCHANGE_CLIENTID: self._home.apExchangeClientId,
+            ATTR_APEXCHANGE_STATE: self._home.apExchangeState,
+            }
+
+
+class HomematicipDeviceStatus(HomematicipGenericDevice):
+    """Representation of an HomematicIP sensor device."""
+
+    def __init__(self, hass, home, device, signal=None):
+        """Initialize the device."""
+        super().__init__(hass, home, device)
+        _LOGGER.debug('Setting up sensor device status: %s',
+                      device.label)
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name('Status')
+
+    @property
+    def icon(self):
+        """Return the icon of the generic device."""
+        if self._device.unreach is None or self._device.unreach:
+            return 'mdi:wifi-off'
```

Comment:
Do not make the icon static. Use `available` property for unreachable. ATTR_BATTERY_LEVEL for battery info.